Sources/AblyChat/MessageReaction+JSON.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+
+internal extension MessageReactionSummary {
+    enum JSONKey: String {
+        case messageSerial
+        case unique
+        case distinct
+        case multiple
+    }
+
+    init(messageSerial: String, values jsonObject: [String: JSONValue]) throws(InternalError) {
+        self.messageSerial = messageSerial
+
+        let uniqueJson = try? jsonObject.optionalObjectValueForKey(MessageReactionType.unique.rawValue) ??
```

Comment:
1. Why do we check under two different keys?
2. I would suggest that instead of doing this parsing here, we implement the `SummaryDistinctV1` etc helper types in ably-cocoa (https://sdk.ably.com/builds/ably/specification/main/features/#TM7b). (they don't need to be static methods on `ARTMessage`, they can just be top-level classes).

Reply:
I've left the same link to the internal [discussion](https://ably-real-time.slack.com/archives/C02NY1VT3LY/p1749924228762039?thread_ts=1749655305.091679&cid=C02NY1VT3LY) in the code comment.

Reply:
Have poked there to see what's the latest